ipam/ring/ring.go

```diff
@@ -19,8 +19,14 @@ import (
 	"github.com/weaveworks/weave/router"
 )
 
+const MaxClockSkew int64 = int64(time.Hour)
+
+// Hook for replacing for testing
+var Now = func() int64 { return time.Now().Unix() }
```

Comment:
I'll unexport it (and the maxclockskew const)

On Thu, Apr 23, 2015 at 3:19 PM Bryan Boreham <notifications@github.com>
wrote:

> In ipam/ring/ring.go
> <https://github.com/bboreham/weave/pull/13#discussion_r28966242>:
>
> > @@ -19,8 +19,14 @@ import (
> >  	"github.com/weaveworks/weave/router"
> >  )
> >
> > +const MaxClockSkew int64 = int64(time.Hour)
> > +
> > +// Hook for replacing for testing
> > +var Now = func() int64 { return time.Now().Unix() }
>
> This could be non-exported, given current usage. Or, if expected to be
> called in a test crossing Allocator and Ring, perhaps both should share the
> same time source?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bboreham/weave/pull/13/files#r28966242>.
>